// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

edition = "2023";

package optimization_guide.proto;

import "components/optimization_guide/proto/features/common_quality_data.proto";

import "components/optimization_guide/proto/model_quality_metadata.proto";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.optimization_guide.features.proto";

option java_outer_classname = "ZeroStateSuggestionsProto";

// DO NOT EDIT THIS FILE DIRECTLY!
//
// This file is generated in g3 and then synced to Chrome. Instead, please refer to
// http://go/chrome-mqls-onboarding (Google-internal link), and then changes will
// be synced with Chrome automatically.

message ZeroStateSuggestionsLoggingData {
  // The feature request.
  ZeroStateSuggestionsRequest request = 1 [features = { field_presence: EXPLICIT }];

  // The feature response.
  ZeroStateSuggestionsResponse response = 2 [features = { field_presence: EXPLICIT }];

  ModelExecutionInfo model_execution_info = 3 [features = { field_presence: EXPLICIT }];
}

message ZeroStateSuggestionsRequest {
  // The context of the page to generate suggestions for.
  PageContext page_context = 1 [features = { field_presence: EXPLICIT }];

  // Whether the request is for FRE mode.
  bool is_fre = 2 [features = { field_presence: EXPLICIT }];
}

message ZeroStateSuggestionsResponse {
  repeated Suggestion suggestions = 1;
}

message Suggestion {
  string label = 1 [features = { field_presence: EXPLICIT }];
}
